app/Http/routes.php

```diff
@@ -47,23 +47,19 @@
 
 Route::get('/icons', 'HomeController@getIcons');
 
-// beatmaps section
-Route::get('/beatmaps', ['as' => 'beatmaps', 'uses' => 'BeatmapController@index']);
-Route::get('/beatmaps/search/{filters?}', ['as' => 'beatmaps.search', 'uses' => 'BeatmapController@search']);
-
-// maps
-Route::get('/beatmaps/set/{id}', ['as' => 'set', 'uses' => 'BeatmapController@getMapSet']);
-Route::get('/beatmaps/map/{id}', ['as' => 'beatmap', 'uses' => 'BeatmapController@getMap']);
-Route::get('/beatmaps/modding/{id?}', ['as' => 'modding', 'uses' => 'ModdingController@getModding']);
-Route::get('/beatmaps/packs', ['as' => 'packs', 'uses' => 'BeatmapController@getPacks']);
-Route::get('/beatmaps/charts/{id?}', ['as' => 'charts', 'uses' => 'BeatmapController@getCharts']);
-
-Route::get('/b/{id}', ['as' => 'beatmaps.show', function ($id) {
-    return Redirect::to('https://osu.ppy.sh/b/'.$id);
-}]);
+// Route::get('/beatmaps/packs', ['as' => 'packs', 'uses' => 'BeatmapController@getPacks']);
+// Route::get('/beatmaps/charts/{id?}', ['as' => 'charts', 'uses' => 'BeatmapController@getCharts']);
+
+Route::resource('/beatmaps', 'BeatmapController', ['only' => ['show']]);
+Route::get('/beatmaps/{id}/scores', ['as' => 'beatmap.scores', 'uses' => 'BeatmapController@getScores']);
+
+Route::resource('/beatmapsets', 'BeatmapsetsController', ['only' => ['index', 'show']]);
+Route::get('/beatmapset-search/{filters?}', ['as' => 'beatmaps.search', 'uses' => 'BeatmapsetsController@search']);
```

Comment:
Wouldn't it get routed to `BeatmapsetsController@show` instead of `search` in this case,